### PR TITLE
SO-4712: Commit classification changes in commit water mark low batches

### PR DESCRIPTION
**Pull request by AAAlinaaa**

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java [hunk 1]
```diff
@@ -66,12 +70,23 @@
 import com.b2international.snowowl.snomed.datastore.request.SnomedRelationshipUpdateRequestBuilder;
 import com.b2international.snowowl.snomed.datastore.request.SnomedRequests;
 import com.b2international.snowowl.snomed.reasoner.classification.ClassificationTracker;
-import com.b2international.snowowl.snomed.reasoner.domain.*;
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java:69]: Revert this change please and configure your IDE to use wildcard import when the number of import statements is greater than or equal to `5`.

File: snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java [hunk 2]
```diff
@@ -200,26 +215,42 @@ private Boolean persistChanges(final BranchContext context, final IProgressMonit
...
+					.setAuthor(userId)
+					.build();
+			
+			final CommitResult commitResult = new IdRequest<>(commitRequest).execute(context);
+			resultTimeStamp = commitResult.getCommitTimestamp();
+		}
 
-		classificationTracker.classificationSaved(classificationId, commitResult.getCommitTimestamp());
-		return Boolean.TRUE;
+		if (EffectiveTimes.UNSET_EFFECTIVE_TIME == resultTimeStamp) {
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java]: Why `EffectiveTimes.UNSET_EFFECTIVE_TIME`? That constant is being used for other purposes. Please use `Commit.NO_COMMIT_TIMESTAMP` instead.

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java [hunk 3]
```diff
@@ -200,26 +196,42 @@ private Boolean persistChanges(final BranchContext context, final IProgressMonit
 		classificationTracker.classificationSaving(classificationId);
 
 		final SubMonitor subMonitor = SubMonitor.convert(monitor, "Persisting changes", 6);
-		final BulkRequestBuilder<TransactionContext> bulkRequestBuilder = BulkRequest.create();
+		final List<RequestBuilder<TransactionContext, ?>> bulkRequestBuilder = Lists.newArrayList();
```
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.reasoner/src/com/b2international/snowowl/snomed/reasoner/request/SaveJobRequest.java:199]: You can make this change a lot smaller by skipping all `List<RequestBuilder<TransactionContext, ?>>` changes and `(RequestBuilder<TransactionContext, ?>)` casts below, if you keep the `BulkRequestBuilder` type here, then call `build` on it and ask for its request list via `BulkRequest#getRequests` for the partitioning step.

**Review by apeteri — CHANGES_REQUESTED** [review]

**Comment by AAAlinaaa** [comment]
Closed in favor of https://github.com/b2ihealthcare/snow-owl/pull/809

**Pull request closed without merging**